wrappers/hnynethttp/nethttp.go

```diff
@@ -156,7 +158,23 @@ func (ht *hnyTripper) spanRoundTrip(ctx context.Context, span *trace.Span, r *ht
 	}
 	span.AddField("meta.type", "http_client")
 	span.AddField("name", "http_client")
-	r.Header.Add(propagation.TracePropagationHTTPHeader, span.SerializeHeaders())
+
+	otelSpan := trace.OTelSpanFromContext(ctx)
+	ctx = oteltrace.ContextWithSpan(ctx, otelSpan)
```

Comment:
This feels out of place here- shouldn't this be in a propagator?

Reply:
Oooh, yeah they should be. The reason they're here is because the W3C trace context propagator in the otel sdk expects an otel span, so we grab it here and shove it into the context before calling the Inject method on each of the propagators, one of which could be the W3C propagator in the otel sdk. 

This is making me think it'd be better to wrap that propagator. Unfortunately, that'd mean wrapping all of the otel propagators, which feels brittle. Open to thoughts - better to just comment here, or move these into a wrapped propagator?

Reply:
Opted to keep as-is, but with better commenting.

Reply:
With the most recent commit, this is gone!